homeassistant/components/reolink/host.py

```diff
@@ -386,10 +430,40 @@ def unregister_webhook(self):
         webhook.async_unregister(self._hass, self.webhook_id)
         self.webhook_id = None
 
+    async def _async_long_polling(self, *_) -> None:
+        """Use ONVIF long polling to immediately receive events."""
+        # This task will be cancelled once _async_stop_long_polling is called
+        while True:
+            if self._webhook_reachable:
+                self._long_poll_task = None
+                await self._async_stop_long_polling()
+                return
+
+            try:
+                channels = await self._api.pull_point_request()
+            except Exception as ex:  # pylint: disable=broad-except
+                if not self._long_poll_error:
+                    _LOGGER.exception("Error while requesting ONVIF pull point: %s", ex)
+                    await self._api.unsubscribe(sub_type=SubType.long_poll)
+                self._long_poll_error = True
+                await asyncio.sleep(30)
```

Comment:
I would make 30 a named constant to show the intent here

Reply:
done